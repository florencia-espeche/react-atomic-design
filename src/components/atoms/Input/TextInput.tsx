import React, { ChangeEventHandler, HTMLAttributes } from 'react'
import { StyledTextInput, StyledWrapper } from './TextInput.styles'
import { useTheme } from '../../../hooks/useTheme'

export interface TextInputProps
    extends HTMLAttributes<HTMLInputElement> {
    disabled?: boolean
    error?: boolean
    inputSize?: 'large' | 'medium' | 'small'
    value?: string
    width?: string
    onChange?: ChangeEventHandler<HTMLInputElement>
}

export const TextInput: React.ForwardRefRenderFunction<
    HTMLInputElement,
    TextInputProps
> = props => {
    const {
        disabled = false,
        error = false,
        inputSize = 'medium',
        value,
        onChange,
        placeholder,
        width = '100%',
        ...rest
    } = props

    const theme = useTheme()

    const styles = {
        disabled: disabled,
        error,
        width
    }

    return (
        <StyledWrapper inputSize={inputSize} width={width} theme={theme}>
            <StyledTextInput
                data-testid="input"
                value={value}
                onChange={onChange}
                placeholder={placeholder}
                theme={theme}
                {...styles}
                {...rest}
            />
        </StyledWrapper>
    )
}
