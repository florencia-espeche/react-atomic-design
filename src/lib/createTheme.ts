import { defaultTheme } from '../styles/defaultTheme'
import { Theme } from '../styles/types'

export const createTheme = <T>(theme: Theme & T) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },
        borderRadius: theme.borderRadius || defaultTheme.borderRadius,
        fontTokens: {
            ...defaultTheme.fontTokens,
            ...theme.fontTokens
        },
        fontWeight: theme.fontWeight || defaultTheme.fontWeight,
        name: theme.name || defaultTheme.name
    }
}
