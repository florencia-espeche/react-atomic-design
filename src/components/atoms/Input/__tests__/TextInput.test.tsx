import React from 'react'
import { TextInput } from '../TextInput'
import { fireEvent, render } from '../../../../lib/test-utils'
import 'jest-styled-components'

describe('TextInput Component', () => {
    it('Should be able to match a snapshot', () => {
        const { container } = render(<TextInput placeholder="Test" />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should render the component with medium size', () => {
        const { container } = render(
            <TextInput placeholder="Test" inputSize="medium" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to match a snapshot with true parameter', () => {
        const { container } = render(
            <TextInput placeholder="Test" error={false} />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to change a value of input', () => {
        const { getByTestId } = render(
            <TextInput
                error={false}
                placeholder="placeholder"
                onChange={jest.fn}
            />
        )
        const input = getByTestId('input')
        const value = 'value'
        fireEvent.change(input, { target: { value } })
        expect((input as HTMLInputElement).value).toBe(value)
    })
})
