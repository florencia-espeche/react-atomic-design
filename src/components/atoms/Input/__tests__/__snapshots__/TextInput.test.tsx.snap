// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component Should be able to match a snapshot 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: 32px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  background-color: #EEEEEE;
  padding: 0;
  width: 100%;
  border: none;
  color: #000;
  border-radius: 0;
  outline: none;
  -webkit-transition: 0.1s ease-out;
  transition: 0.1s ease-out;
  box-shadow: inset 0 0 0 2px transparent;
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #000;
}

<div
  class="c0"
  width="100%"
>
  <input
    class="c1"
    data-testid="input"
    placeholder="Test"
    value=""
    width="100%"
  />
</div>
`;

exports[`TextInput Component Should be able to match a snapshot with true parameter 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: 32px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  background-color: #EEEEEE;
  padding: 0;
  width: 100%;
  border: none;
  color: #000;
  border-radius: 0;
  outline: none;
  -webkit-transition: 0.1s ease-out;
  transition: 0.1s ease-out;
  box-shadow: inset 0 0 0 2px transparent;
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #000;
}

<div
  class="c0"
  width="100%"
>
  <input
    class="c1"
    data-testid="input"
    placeholder="Test"
    value=""
    width="100%"
  />
</div>
`;

exports[`TextInput Component Should render the component with medium size 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: 32px;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  background-color: #EEEEEE;
  padding: 0;
  width: 100%;
  border: none;
  color: #000;
  border-radius: 0;
  outline: none;
  -webkit-transition: 0.1s ease-out;
  transition: 0.1s ease-out;
  box-shadow: inset 0 0 0 2px transparent;
}

.c1:focus {
  box-shadow: inset 0 0 0 2px #000;
}

<div
  class="c0"
  width="100%"
>
  <input
    class="c1"
    data-testid="input"
    placeholder="Test"
    value=""
    width="100%"
  />
</div>
`;
